ports/nrf/common-hal/_bleio/Adapter.c

```diff
@@ -669,6 +715,27 @@ void common_hal_bleio_adapter_start_advertising(bleio_adapter_obj_t *self, bool
     if (advertising_data_bufinfo->len > 31 && scan_response_data_bufinfo->len > 0) {
         mp_raise_bleio_BluetoothError(translate("Extended advertisements with scan response not supported."));
     }
+
+    // Anonymous mode requires a timeout so that we don't continue to broadcast
+    // the same data while cycling the MAC address -- otherwise, what's the
+    // point of randomizing the MAC address?
+    if (!timeout) {
+        if (anonymous) {
+            // The Nordic macro is in units of 10ms. Convert to seconds.
+            uint32_t adv_timeout_max_secs = BLE_GAP_ADV_TIMEOUT_LIMITED_MAX / 100;
```

Comment:
You could add a `UNITS_TO_SEC` macro to `ble_drv.h` to do the reverse of `SEC_TO_UNITS`.